src/drawing/DrawingCreator.tsx

```diff
@@ -0,0 +1,163 @@
+        return [((x - left) / width) * 100, ((y - top) / height) * 100];
+    };
+
+    // Drawing Lifecycle Callbacks
+    const startDraw = (x: number, y: number): void => {
+        const [x1, y1] = getPosition(x, y);
+
+        setDrawingStatus(DrawingStatus.dragging);
+
+        capturedPathRef.current = [{ x: x1, y: y1 }];
+        drawingDirtyRef.current = true;
+    };
+    const stopDraw = (): void => {
+        const pathGroup = getPathGroup();
+
+        setDrawingStatus(DrawingStatus.init);
+
+        capturedPathRef.current = [];
+        drawingDirtyRef.current = true;
+
+        if (pathGroup) {
+            onStop(pathGroup);
+        } else {
+            onAbort();
+        }
+    };
+    const updateDraw = (x: number, y: number): void => {
+        const [x2, y2] = getPosition(x, y);
+        const { current: points } = capturedPathRef;
+
+        points.push({ x: x2, y: y2 });
+        drawingDirtyRef.current = true;
+
+        if (drawingStatus !== DrawingStatus.drawing) {
+            setDrawingStatus(DrawingStatus.drawing);
+            onStart();
+        }
+    };
+
+    // Event Handlers
+    const renderStep = (callback: () => void): void => {
+        renderHandleRef.current = window.requestAnimationFrame(callback);
+    };
+    const renderPath = (): void => {
+        const { current: creatorEl } = creatorElRef;
+        const { current: points } = capturedPathRef;
+        const { current: isDirty } = drawingDirtyRef;
+        const { current: svgPath } = drawingPathRef;
+
+        if (!creatorEl) {
+            return;
+        }
+
+        const d = getPathCommands(points);
+
+        if (isDirty && svgPath && d.length) {
+            svgPath.setAttribute('d', d);
+
+            drawingDirtyRef.current = false;
+        }
+
+        renderStep(renderPath);
+    };
+
+    React.useEffect(() => {
+        if (drawingStatus !== DrawingStatus.init) {
+            renderStep(renderPath);
+        }
+
+        return () => {
+            const { current: renderHandle } = renderHandleRef;
+
+            // Cancel the render loop
+            if (renderHandle) {
+                window.cancelAnimationFrame(renderHandle);
+            }
+        };
+    }, [drawingStatus]); // eslint-disable-line react-hooks/exhaustive-deps
+
+    const { strokeWidth } = getStrokeWidths(stroke.size, drawingSVGRef.current) ?? {};
+
+    return (
+        // eslint-disable-next-line jsx-a11y/mouse-events-have-key-events
+        <PointerCapture
+            ref={creatorElRef}
+            className={classNames(className, 'ba-DrawingCreator')}
+            data-testid="ba-DrawingCreator"
+            onDrawStart={startDraw}
+            onDrawStop={stopDraw}
+            onDrawUpdate={updateDraw}
+            status={drawingStatus}
+        >
+            <DrawingSVG ref={drawingSVGRef} className="ba-DrawingCreator-current">
+                <g fill="transparent" stroke={stroke.color} strokeWidth={strokeWidth}>
```

Comment:
Can we use `DrawingPathGroup` directly here rather than importing `getStrokeWidths`? We should strive to use the components themselves rather than their inner logic when possible.